### PR TITLE
[SDPAP-8093] remove deprecated modules before d10 update

### PR DESCRIPTION
**Pull request by MdNadimHossain**
### Jira
https://digital-vic.atlassian.net/browse/SDPAP-8093

### Problem/Motivation
Some of the module was deprecated in Drupal 9.4 and the code package reference is removed from Drupal 10. So those modules needs to be uninstalled first from SDP projects before installing the Drupal 10

### Fix
1. Removed Deprecated modules in 9.4 and added update hook to clean it from all existing projects.
2.

### CODE REVIEW

File: tide_core.install [hunk 1]
```diff
@@ -1914,3 +1914,23 @@ function tide_core_update_8090() {
...
+/**
+ * Remove deprecated modules before d10 upgrade.
+ */
+function tide_core_update_8091() {
+  $modules = [
+    'ckeditor',
+    'wysiwyg_template_core',
+    'color',
+    'hal',
+    'tide_spell_checker',
```
Comment (MdNadimHossain) [inline tide_core.install:1927]: `tide_spell_checker` Will not be supported anymore and the code will be cleaned up in the Drupal 10 upgrade

**Comment by krakerag** [comment]
Do we need to remove them out of composer.json also for dependency checking?

**Comment by MdNadimHossain** [comment]
> Do we need to remove them out of composer.json also for dependency checking?

It is removed in the core composer of d10. That is why we need to uninstall it first before updating the projects to d10. Else it will complain the package does not exist. So all these are marked as deprecated from 9.4, so first step will be uninstalling them from the cms while the package is still there. In d10 they are removed, so it will get cleaned up from the composer during the d10 upgrade

**Review by krakerag — APPROVED** [review]

**Comment by anthony-malkoun** [comment]
> Do we need to remove them out of composer.json also for dependency checking?

The only problem with that is it has to be a two step process. The uninstall requires the code to be there.

**Pull request merged**